ui/src/chatbot.tsx

```diff
@@ -110,29 +107,29 @@ export const XChatbot = ({ model }: { model: Chatbot }) => {
         {msgs.map(({ msg, fromUser }, idx) => (
           <div
             key={idx}
-            className={css.msgWrapper}
+            className={clas(css.msgWrapper, fromUser ? css.userMsg : '')}
             style={{
-              marginTop: msgs[idx - 1]?.fromUser !== fromUser ? 3 : 10,
-              textAlign: fromUser ? 'right' : 'left',
-              color: getContrast(fromUser ? '$themePrimary' : '$text')
+              paddingTop: msgs[idx - 1]?.fromUser !== fromUser ? rem(0.7) : 0,
+              paddingBottom: msgs?.[idx + 1]?.fromUser !== fromUser ? rem(0.85) : 0,
+              color: cssVar(fromUser ? '$neutralDark' : '$themePrimary')
```

Comment:
Let's stick with a single text color that is easy to read. I would maybe even consider using `getContrast` function to be sure the text is properly visible in all cases.
```suggestion
              color: cssVar('$neutralDark')
```

Reply:
A problem with `getContrast` is that it does not work dynamically when switching themes.

But I agree with the single easy to read text color.

Reply:
> A problem with getContrast is that it does not work dynamically when switching themes.

Wouldn't a rerender of the component on the theme change be enough?